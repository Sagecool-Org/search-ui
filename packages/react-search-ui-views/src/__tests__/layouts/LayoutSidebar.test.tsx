import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import LayoutSidebar from "../../layouts/LayoutSidebar";

it("renders correctly", () => {
  const { container } = render(
    <LayoutSidebar className="sui-layout-sidebar">Hello world!</LayoutSidebar>
  );
  expect(container).toMatchSnapshot();
});

it("renders toggled class based on state", () => {
  const { container } = render(
    <LayoutSidebar className="sui-layout-sidebar">Hello world!</LayoutSidebar>
  );

  const toggleButton = screen.getByText("Show Filters");
  fireEvent.click(toggleButton);

  const sidebar = container.querySelector(".sui-layout-sidebar");
  expect(sidebar).toHaveClass("sui-layout-sidebar--toggled");
});

it("updates isSidebarToggled state on button click", () => {
  const { container } = render(
    <LayoutSidebar className="sui-layout-sidebar">Hello world!</LayoutSidebar>
  );

  const toggleButton = screen.getByText("Show Filters");
  const sidebar = container.querySelector(".sui-layout-sidebar");

  expect(sidebar).not.toHaveClass("sui-layout-sidebar--toggled");

  fireEvent.click(toggleButton);
  expect(sidebar).toHaveClass("sui-layout-sidebar--toggled");

  fireEvent.click(toggleButton);
  expect(sidebar).not.toHaveClass("sui-layout-sidebar--toggled");
});
