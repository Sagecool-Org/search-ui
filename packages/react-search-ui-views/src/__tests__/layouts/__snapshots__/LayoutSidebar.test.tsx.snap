// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    class="sui-layout-sidebar-toggle"
    hidden=""
    type="button"
  >
    Show Filters
  </button>
  <div
    class="sui-layout-sidebar"
  >
    <button
      class="sui-layout-sidebar-toggle"
      hidden=""
      type="button"
    >
      Save Filters
    </button>
    Hello world!
  </div>
</div>
`;
